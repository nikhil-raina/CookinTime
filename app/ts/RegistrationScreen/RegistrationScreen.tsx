/* eslint-disable prettier/prettier */
/* eslint-disable react-native/no-inline-styles */
/* eslint-disable prettier/prettier */
/**
 * @providesModule RegistrationScreen
 * @flow
 */

import React from 'react';
import styles from './RegistrationScreenStyle';
import images from '../CIT-Modules/ImageTree';
import pogoTheme  from '../../assets/theme/pogo';

import {
  View,
  Image,
  Text,
  ImageBackground,
  TextInput,
  TouchableOpacity,
  Keyboard,
} from 'react-native';

interface State {
  showFirstScreen: boolean,
  username: string,
  email: string,
  password: string,
  phoneNumber: string,
  passwordAgain: string,
}

export default class RegistrationScreen extends React.Component<{}, State> {

  // TODO: Get the placeholder HEX code for all colors
  constructor(props: any) {
    super(props);
    this.state = {
      showFirstScreen: true,
      email: '',
      password: '',
      username: '',
      phoneNumber: '',
      passwordAgain: '',
    };
  }

  _setEmail(value: string) {
    this.setState({
      email: value,
    });
  }

  _setPassword(value: string) {
    this.setState({
      password: value,
    });
  }

  _setPasswordAgain(text: string) {
    this.setState({
      passwordAgain: text,
    });
  }

  _setUserName(text: string) {
    this.setState({
      username: text,
    });
  }

  _setPhoneNumber(text: string) {
    this.setState({
      phoneNumber: text,
    });
  }

  _firstCheckScreen = () => {
    // TODO: Need to create modals here to show that something was invalid.
    const fail = setTimeout(() => {
      Keyboard.dismiss();
      console.log('being 1 called');
      this.setState({
        email: '',
        password: '',
        passwordAgain: '',
      });
    }, 150);  // The delay is for it to look realistically smooth

    if (!this._isEmailValid()) {
      console.log('Email Not Valid');
      fail;
    } else if (!this._isPasswordSame()) {
      console.log('Password is not the same');
      fail;
    } else {
      console.log('All First is Valid');
      this.setState({
        showFirstScreen: false,
      });
    }
  }

  _secondCheckScreen = () => {
    // TODO: Need to create modals here to show that something was invalid.
    const fail = setTimeout(() => {
      Keyboard.dismiss();
      console.log('being 2 called');
      this.setState({
        email: '',
        password: '',
        passwordAgain: '',
      });
    }, 150);  // The delay is for it to look realistically smooth
    if (!this._isUsernameValid()) {
      console.log('Username Not Valid');
      fail;
    } else if (isNaN(parseInt(this.state.phoneNumber))) {
      console.log('Phone number Not Valid');
      fail;
    } else {
      console.log('All Second is Valid');
      // TODO: Navigate to Main Screen
    }
  }

  _isUsernameValid() {
    // eslint-disable-next-line no-useless-escape
    return !/[~` !#$%\^&*+=\-\[\]\\';,/{}|\\":<>\?]/g.test(this.state.username);
  }

  _isEmailValid() {
    const expression = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return expression.test(this.state.email.toLowerCase());
  }

  _isPasswordSame() {
    return this.state.password.toLowerCase() === this.state.passwordAgain.toLowerCase();
  }

  _renderFirstScreen(): React.ReactElement {
    const emailText = 'E-mail:';
    const pwdText = 'Password:';
    const rePwdText = 'Re-enter Password:';
    return (
      <View style={styles.container}>
        <View style={styles.credentialsContainer}>
          <View style={{alignItems: 'center', marginTop: 80, marginBottom: 55}}>
            <Text style={styles.headerText}>
              {'Sign Up'}
            </Text>
          </View>
          <View style={styles.nameContainer}>
            <Text style={styles.credentialHeaderText}>
              {emailText}
            </Text>
            <View style={styles.replyTextContainer}>
              <TextInput  keyboardType={'email-address'}
                          style={styles.textStyle}
                          autoCompleteType={'email'}
                          underlineColorAndroid={pogoTheme.LIGHT_YELLOW.color}
                          onChangeText={(value) => this._setEmail(value)}
                          placeholder={'Ex. BobBobby@hungry.com'}
                          value={this.state.email}
                          placeholderTextColor={'white'}/>
            </View>
          </View>
          <View style={styles.nameContainer}>
            <Text style={styles.credentialHeaderText}>
              {pwdText}
            </Text>
            <View style={styles.replyTextContainer}>
              <TextInput  keyboardType={'default'}
                          secureTextEntry={true}
                          style={styles.textStyle}
                          underlineColorAndroid={pogoTheme.LIGHT_YELLOW.color}
                          onChangeText={(value) => this._setPassword(value)}
                          value={this.state.password}
                          placeholder={'Be secretive'}
                          placeholderTextColor={'white'}/>
            </View>
          </View>
          <View>
            <Text style={styles.credentialHeaderText}>
              {rePwdText}
            </Text>
            <View style={styles.replyTextContainer}>
              <TextInput  keyboardType={'default'}
                          secureTextEntry={true}
                          style={styles.textStyle}
                          underlineColorAndroid={pogoTheme.LIGHT_YELLOW.color}
                          onChangeText={(value) => this._setPasswordAgain(value)}
                          value={this.state.passwordAgain}
                          placeholder={'Be secretive again'}
                          placeholderTextColor={'white'}/>
            </View>
          </View>
        </View>
        <View>
          <TouchableOpacity onPress={this._firstCheckScreen}>
            <Text style={[styles.textStyle, styles.buttonTextStyle]}>
              {'NEXT'}
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

  _renderSecondScreen(): React.ReactElement {
    const defaultProfilePicture = images.REGISTRATION_DEFAULT_PROFILE_PICTURE;
    const defaultProfilePictureShadow = images.REGISTRATION_DEFAULT_PROFILE_PICTURE_SHADOW;
    const nameText = 'Full name:';
    const contactText = 'Contact Number:';
    return (
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          {/* TODO: Make the image clickable so that one could add an image to their profile and edit it. */}
          <ImageBackground  source={defaultProfilePictureShadow}
                            style={styles.logoStyleShadow}>
            <Image  source={defaultProfilePicture}
                    style={styles.logoStyle}/>
          </ImageBackground>
        </View>
        <View style={styles.credentialsContainer}>
          <View style={styles.nameContainer}>
            <Text style={styles.credentialHeaderText}>
              {nameText}
            </Text>
            <View style={styles.replyTextContainer}>
              <TextInput  keyboardType={'default'}
                          style={styles.textStyle}
                          underlineColorAndroid={pogoTheme.LIGHT_YELLOW.color}
                          onChangeText={(value) => this._setUserName(value)}
                          value={this.state.username}
                          placeholder={'Ex. Bob Bobby'}
                          placeholderTextColor={'white'}/>
            </View>
          </View>
          <View>
            <Text style={styles.credentialHeaderText}>
              {contactText}
            </Text>
            <View style={styles.replyTextContainer}>
              <TextInput  keyboardType={'numeric'}
                          style={styles.textStyle}
                          underlineColorAndroid={pogoTheme.LIGHT_YELLOW.color}
                          onChangeText={(value) => this._setPhoneNumber(value)}
                          value={this.state.phoneNumber}
                          placeholder={'123456789'}
                          placeholderTextColor={'white'}/>
            </View>
          </View>
        </View>
        <View>
          <TouchableOpacity onPress={this._secondCheckScreen}>
            <Text style={[styles.textStyle, styles.buttonTextStyle]}>
              {'ENTER'}
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }


  render(): React.ReactElement {
    const background_image = images.REGISTRATION_BACKGROUND;
    return (
      <ImageBackground  source={background_image}
                        style={styles.backgroundImage}>
        {this.state.showFirstScreen ?
          this._renderFirstScreen() :
          this._renderSecondScreen()}
      </ImageBackground>
    );
  }
}
