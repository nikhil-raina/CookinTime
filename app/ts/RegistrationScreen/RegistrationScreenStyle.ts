/* eslint-disable prettier/prettier */
/**
 * @provideModule RegistrationScreenStyle
 * @flow
 */

import pogoTheme  from '../../assets/theme/pogo';
import tronTheme from '../../assets/theme/tron';
import {
  StyleSheet,
} from 'react-native';


export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },

  backgroundImage: {
    ...tronTheme.defaultBackgroundImage,
  },

  logoStyle: {
    height: 140,
    width: 140,
  },

  logoStyleShadow: {
    alignItems: 'center',
    height: 160,
    width: 150,
  },

  imageContainer: {
    marginTop: 70,
    marginBottom: 30,
  },

  credentialsContainer: {
    marginBottom: 40,
    width: 280,
  },

  replyTextContainer: {
    paddingLeft: 5,
  },

  credentialHeaderText: {
    ...pogoTheme.YELLOW,
    ...tronTheme.font.light,
    ...tronTheme.text.primaryText,
  },

  headerText: {
    ...pogoTheme.YELLOW,
    ...tronTheme.font.reg,
    ...tronTheme.text.ternaryHeader,
  },

  nameContainer: {
    marginBottom: 25,
  },

  textStyle: {
    ...tronTheme.font.reg,
    ...tronTheme.text.subHeaderDetail,
    ...pogoTheme.YELLOW,
  },

  buttonTextStyle: {
    borderBottomWidth: 1,
    borderBottomColor: pogoTheme.LIGHT_YELLOW.color,
    ...tronTheme.text.subHeader,
    ...tronTheme.font.semiBold,
  },
});
