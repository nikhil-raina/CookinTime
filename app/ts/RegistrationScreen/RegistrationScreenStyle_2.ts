/* eslint-disable prettier/prettier */
/**
 * @provideModule RegistrationScreenStyle_2
 * @flow
 */

import {
  StyleSheet,
  Dimensions,
} from 'react-native';


export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },

  backgroundImage: {
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    backgroundColor: 'rgba(0,0,0,0.6)',
  },

  logoStyle: {
    height: 140,
    width: 140,
  },

  imageContainer: {
    marginTop: 70,
    marginBottom: 30,
  },

  credentialsContainer: {
    marginBottom: 40,
    width: 280,
  },

  replyTextContainer: {
    paddingLeft: 5,
  },

  credentialHeaderText: {
    color: '#F2D750',
    fontFamily: 'JosefinSans-Light',
    fontSize: 14,

  },

  nameContainer: {
    marginBottom: 25,
  },

  textStyle: {
    fontFamily: 'JosefinSans-Regular',
    fontSize: 18,
    color: '#F2E6C2',
    
  },


});
