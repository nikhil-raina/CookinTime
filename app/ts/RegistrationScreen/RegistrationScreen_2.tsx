/* eslint-disable prettier/prettier */
/* eslint-disable react-native/no-inline-styles */
/**
 * @providesModule RegistrationScreen_2
 * @flow
 */

import React from 'react';
import styles from './RegistrationScreenStyle_2';
import images from '../CIT-Modules/ImageTree';

import {
  View,
  Image,
  Text,
  ImageBackground,
  TextInput,
} from 'react-native';

export default class RegistrationScreen_2 extends React.Component<{}, {}> {

  constructor(props: any) {
    super(props);
  }

  render(): React.ReactElement {
    const background_image = images.REGISTRATION_BACKGROUND;
    const defaultProfilePicture = images.REGISTRATION_DEFAULT_PROFILE_PICTURE;
    const defaultProfilePictureShadow = images.REGISTRATION_DEFAULT_PROFILE_PICTURE_SHADOW;
    const nameText = 'Full name:';
    const contactText = 'Contact Number:';
    return (
      <ImageBackground  source={background_image}
                        style={styles.backgroundImage}>
        <View style={styles.container}>
          <View style={styles.imageContainer}>
            <Image  source={defaultProfilePicture}
                    style={styles.logoStyle}/>
          </View>
          <View style={styles.credentialsContainer}>
            <View style={styles.nameContainer}>
              <Text style={styles.credentialHeaderText}>
                {nameText}
              </Text>
              <View style={styles.replyTextContainer}>
                <TextInput  keyboardType={'default'}
                            style={styles.textStyle}
                            underlineColorAndroid={'#F2D750'}
                            value={'Nikhil Raina'}/>
              </View>
            </View>
            <View>
              <Text style={styles.credentialHeaderText}>
                {contactText}
              </Text>
              <View style={styles.replyTextContainer}>
                <TextInput  keyboardType={'numeric'}
                            style={styles.textStyle}
                            underlineColorAndroid={'#F2D750'}
                            value={'1234567890'}/>
              </View>
            </View>
          </View>
          <View>
            <Text style={[styles.textStyle, styles.buttonTextStyle]}>
              {'NEXT'}
            </Text>
          </View>
        </View>
      </ImageBackground>
    );
  }
}
