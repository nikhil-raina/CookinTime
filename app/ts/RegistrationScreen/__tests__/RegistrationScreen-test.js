/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';

import RegistrationScreen from '../RegistrationScreen';

describe('Rendering basic demo screen', () => {
  it('renders correctly', () => {
    const screen = renderer.create(<RegistrationScreen />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
