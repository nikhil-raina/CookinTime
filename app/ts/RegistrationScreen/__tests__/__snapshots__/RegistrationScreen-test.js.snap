// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering basic demo screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.6)",
      "height": 1334,
      "width": 750,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/RegistrationScreen/drawable-hdpi/RegistrationScreen-Background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 30,
          "marginTop": 70,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "height": 160,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/RegistrationScreen/drawable-hdpi/RegistrationScreen-ProfilePictureShadow.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 160,
                "width": 150,
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/RegistrationScreen/RegistrationScreen-ProfilePicture.png",
            }
          }
          style={
            Object {
              "height": 140,
              "width": 140,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 40,
          "width": 280,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 25,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#F2D750",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 14,
            }
          }
        >
          Full name:
        </Text>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#F2E6C2",
                "fontFamily": "JosefinSans-Regular",
                "fontSize": 18,
              }
            }
            underlineColorAndroid="#F2E6C2"
            value="Nikhil Raina"
          />
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#F2D750",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 14,
            }
          }
        >
          Contact Number:
        </Text>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#F2E6C2",
                "fontFamily": "JosefinSans-Regular",
                "fontSize": 18,
              }
            }
            underlineColorAndroid="#F2E6C2"
            value="1234567890"
          />
        </View>
      </View>
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Regular",
              "fontSize": 18,
            },
            Object {
              "borderBottomColor": "#F2E6C2",
              "borderBottomWidth": 1,
              "fontFamily": "JosefinSans-SemiBold",
              "fontSize": 20,
            },
          ]
        }
      >
        ENTER
      </Text>
    </View>
  </View>
</View>
`;
